Composer/packages/client/src/utils/hooks.ts

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { useState, useEffect } from 'react';
+import { globalHistory } from '@reach/router';
+
+export const useLocation = () => {
+  const { location, navigate } = globalHistory;
+  const [state, setState] = useState({ location, navigate });
+  useEffect(() => globalHistory.listen(({ location }) => setState(state => ({ ...state, location }))), []);
```

Comment:
Should this re-use the history listener somehow? As I understand this currently, each component that uses this hook will register their own listener.

Also, does `globalHistory.listen` return a function to remove its listener?

Reply:
I'll look into reusing the listener, and yes, `globalHistory.listen` returns a remove listener function which should be returned in the arrow function and called when the component unmounts. I'll make that clearer.